lib/rspec/support/differ.rb

```diff
@@ -73,6 +84,14 @@ def initialize(opts={})
 
     private
 
+      def hash_with_anything?(arg)
+        Hash === arg && safely_flatten(arg).any? { |a| RSpec::Mocks::ArgumentMatchers::AnyArgMatcher === a }
+      end
+
+      def no_procs_and_no_numbers?(*args)
```

Comment:
I’m also struggling to understand how this works.
 We do pass expected and actual there, and they end up wrapped in an array when flatten is called on it?

Reply:
Indeed... that's weird... What if I remove the star operator from `#no_procs` and `no_numbers` method?

The purpose of this `no_procs_and_no_numbers?` method is to reduce the number of `&&` operators used in the `#diff` method, thus appeasing rubocop by removing the `Metrics::PerceivedComplexity` offense.